### PR TITLE
ROC-7566, ROC-7570: Fully implement hearing court selection

### PR DESCRIPTION
**Pull request by daveoh**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/ROC-7566
https://tools.hmcts.net/jira/browse/ROC-7570

### Change description ###
Fully implement hearing court selection

**Does this PR introduce a breaking change?** (check one with "x")

- [ ] Yes
- [ ] No

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/pilotcourt/PilotCourtService.java [hunk 1]
```diff
@@ -199,7 +202,7 @@ public String getPilotCourtId(HearingCourt hearingCourt) {
 
         return pilotCourts.keySet()
             .stream()
-            .filter(pilotCourtId ->  getPilotHearingCourt(pilotCourtId).equals(hearingCourt))
+            .filter(pilotCourtId ->  getPilotHearingCourt(pilotCourtId).get().equals(hearingCourt))
```
Comment (karthikkannankainos) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/pilotcourt/PilotCourtService.java]: .get() may have trouble passing a gate. Pls verify.

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/OrderPostProcessor.java  2
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/legaladvisor/DocAssemblyTemplateBodyMapper.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=5056965&bid=16645990)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/legaladvisor/DocAssemblyTemplateBodyMapper.java  2
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/OrderPostProcessor.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=5056965&bid=16645990)

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/OrderPostProcessor.java [hunk 2]
```diff
@@ -81,16 +80,39 @@ public CallbackResponse copyDraftToCaseDocument(CallbackParams callbackParams) {
...
+
         return AboutToStartOrSubmitCallbackResponse
             .builder()
-            .data(caseDetailsConverter.convertToMap(cleanUpDynamicList(ccdCase)))
+            .data(caseDetailsConverter.convertToMap(cleanUpDynamicList(updatedCase)))
             .build();
     }
 
+    private HearingCourt getHearingCourt(CCDCase ccdCase) {
+        if (StringUtils.isAllBlank(ccdCase.getHearingCourt())
```
Comment (sabahirfan) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/OrderPostProcessor.java]: As `hearingCourt` is not array will prefer to use `isBlank` insread

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/legaladvisor/DocAssemblyTemplateBodyMapper.java [hunk 3]
```diff
@@ -103,4 +102,20 @@ public DocAssemblyTemplateBody from(CCDCase ccdCase, UserDetails userDetails) {
                 ccdCase.getExpertReportPermissionPartyGivenToDefendant() == YES)
             .build();
     }
+
+    private HearingCourt getHearingCourt(CCDCase ccdCase) {
```
Comment (sabahirfan) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/legaladvisor/DocAssemblyTemplateBodyMapper.java]: This method is duplicated in two different classes, please extract it to some common place and add unit tests.

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/pilotcourt/PilotCourtService.java (hunk 1 above)
Comment (daveoh) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/pilotcourt/PilotCourtService.java]: It passes :)

Comment (sabahirfan) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/pilotcourt/PilotCourtService.java]: How about?
```
getPilotHearingCourt(pilotCourtId).filter(hearingCourt::equals).isPresent())
```

Comment (sabahirfan) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/pilotcourt/PilotCourtService.java]: if `getPilotHearingCourt(pilotCourtI` going to always return an non empty object, then no need to wrap it in optional. Otherwise .get() call is dangerous.

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/DirectionOrderService.java  3
- src/test/java/uk/gov/hmcts/cmc/claimstore/DirectionOrderServiceTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=5056965&bid=16645990)

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/pilotcourt/PilotCourtService.java [hunk 4]
```diff
@@ -47,6 +47,8 @@
     private final String dataSource;
     private final AppInsights appInsights;
 
+    public static final String OTHER_COURT_ID = "OTHER";
```
Comment (sabahirfan) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/pilotcourt/PilotCourtService.java]: Is this relevant to string here? If yes please use same const.

https://github.com/hmcts/cmc-claim-store/blob/master/src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/OrderCreator.java#L220

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/DirectionOrderService.java  3
- src/test/java/uk/gov/hmcts/cmc/claimstore/DirectionOrderServiceTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=5056965&bid=16645990)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/DirectionOrderService.java  3
- src/test/java/uk/gov/hmcts/cmc/claimstore/DirectionOrderServiceTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=5056965&bid=16645990)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/DirectionOrderService.java  3
- src/test/java/uk/gov/hmcts/cmc/claimstore/DirectionOrderServiceTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=5056965&bid=16645990)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/DirectionOrderServiceTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/DirectionOrderService.java  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=5056965&bid=16645990)
